jwst/cube_build/cube_cloud.py

```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
     match
 
-
     """
 
 #________________________________________________________________________________
     if self.instrument == 'MIRI':
-        file_no = file_slice_no
-        det2ab_transform = input_model.meta.wcs.get_transform('detector','alpha_beta')
-        detector2v23 = input_model.meta.wcs.get_transform('detector', 'v2v3')
-        v23toworld = input_model.meta.wcs.get_transform("v2v3","world")
-        worldtov23 = input_model.meta.wcs.get_transform("world","v2v3")
-        v2ab_transform = input_model.meta.wcs.get_transform('v2v3',
-                                                            'alpha_beta')
-
-        alpha, beta, wave = det2ab_transform(x, y)
-        v2, v3, lam23 = detector2v23(x, y)
-        ra,dec,lam = v23toworld(v2,v3,lam23)
-
-        valid1 = np.isfinite(v2)
-        valid2 = np.isfinite(v3)
+        coord1, coord2, wave = input_model.meta.wcs(x, y) # for entire detector find  ra,dec,lambda
+        valid1 = ~np.isnan(coord1)
 
+# MIRI PSF weighting
+# weighting based on distance in alpha-beta plane
         if self.weighting == 'miripsf':
-            wave_resol = self.instrument_info.Get_RP_ave_Wave(this_par1,this_par2)
+            # get transforms needed for this weighting
+
+            det2ab_transform = input_model.meta.wcs.get_transform('detector', 'alpha_beta')
+            worldtov23 = input_model.meta.wcs.get_transform("world", "v2v3")
+            v2ab_transform = input_model.meta.wcs.get_transform('v2v3',
+                                                                'alpha_beta')
+            coord1, coord2, wave = det2ab_transform(x, y)
+            valid1 = ~np.isnan(coord1)
+            wave_resol = self.instrument_info.Get_RP_ave_Wave(this_par1, this_par2)
             alpha_resol = self.instrument_info.Get_psf_alpha_parameters()
             beta_resol = self.instrument_info.Get_psf_beta_parameters()
 
             # transform Cube Spaxel centers to alpha,beta of exposure
-            # for MIRI weighting parameters are based on distance in 
+            # for MIRI weighting parameters are based on distance in
             # alpha-beta coord system
             # transform the cube coordinate values to alpha and beta values
             # xi,eta -> ra,dec
             # ra-dec -> v2,v3
             # v2,v3 -> local alpha,beta
 
-    elif self.instrument == 'NIRSPEC':
+        if self.coord_system == 'alpha-beta': # making Alpha-Beta final Cubes
+            det2ab_transform = input_model.meta.wcs.get_transform('detector',
+                                                                  'alpha_beta')
+            coord1, coord2, wave = det2ab_transform(x, y)
+            valid1 = ~np.isnan(coord1)
+#________________________________________________________________________________
+    elif self.instrument == 'NIRSPEC': # for NIRSPEC need to look over every slice
+        # Slice No are provided by ifu_cube.py
```

Comment:
I want to separate the MIRI weighting methods and pull out when the weighting is based
on the psf then I think it will be cleaner. I decided to wait and just get the pep8 updates done
first.

Reply:
That's fine.